cmd/console/deploy/deploy.go

```diff
@@ -0,0 +1,155 @@
+package deploy
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/davidebianchi/go-jsonclient"
+	"github.com/mia-platform/miactl/renderer"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+type deployConfig struct {
+	Environment         string
+	Revision            string
+	SmartDeploy         bool
+	ForceDeployNoSemVer bool
+}
+
+type deployRequest struct {
+	Environment             string `json:"environment"`
+	Revision                string `json:"revision"`
+	DeployType              string `json:"deployType"`
+	ForceDeployWhenNoSemver bool   `json:"forceDeployWhenNoSemver,omitempty"`
+}
+
+type deployResponse struct {
+	Id  int    `json:"id"`
+	Url string `json:"url"`
+}
+
+func NewDeployCmd() *cobra.Command {
+	var (
+		baseURL   string
+		apiToken  string
+		projectId string
+	)
+
+	cfg := deployConfig{}
+
+	cmd := &cobra.Command{
+		Use:   "deploy",
+		Short: "deploy project",
+		Long:  "trigger the deploy pipeline for selected project",
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			baseURL = viper.GetString("apibaseurl")
+			apiToken = viper.GetString("apitoken")
+			projectId = viper.GetString("project")
+
+			if apiToken == "" {
+				return errors.New("missing API token - please login")
+			}
+			if projectId == "" {
+				cmd.MarkFlagRequired("project")
+			}
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if baseURL == "" {
+				return errors.New("API base URL not specified nor configured")
+			}
+			if projectId == "" {
+				return errors.New("project id not specified nor configured")
+			}
+			r := renderer.New(cmd.OutOrStdout())
+
+			deployData, err := deploy(baseURL, apiToken, projectId, &cfg)
+			if err != nil {
+				r.Error(err).Render()
+				return nil
+			}
+
+			// store triggered pipelines details to enabling checking their status
+			var pipelines Pipelines
+			if err := viper.UnmarshalKey(triggeredPipelinesKey, &pipelines); err != nil {
+				return err
+			}
+
+			viper.Set(triggeredPipelinesKey, append(pipelines, Pipeline{
+				ProjectId:   projectId,
+				PipelineId:  deployData.Id,
+				Environment: cfg.Environment,
+			}))
+			if err := viper.WriteConfig(); err != nil {
+				return err
+			}
+
+			visualizeResponse(r, projectId, deployData)
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVar(&cfg.Environment, "environment", "", "the environment where to deploy the project")
+	cmd.Flags().StringVar(&cfg.Revision, "revision", "", "which version of your project should be released")
+	cfg.SmartDeploy = *cmd.Flags().Bool("smart-deploy", false, "enable smart-deploy feature, which deploys only updated resources")
+	cfg.ForceDeployNoSemVer = *cmd.Flags().Bool("force-no-semver", false, "whether to always deploy pods that do not follow semver")
+
+	cmd.MarkFlagRequired("environment")
+	cmd.MarkFlagRequired("revision")
+
+	return cmd
+}
+
+func deploy(baseUrl, apiToken, projectId string, cfg *deployConfig) (deployResponse, error) {
```

Comment:
Actually I did not understand the purpose of the `sdk` package. Consequently for the moment I decided to try writing commands as independently as possible, so that in the future it will be easy enough to restructure the project code.

Reply:
The sdk package is a sdk to contact the API exposed by the console. It enables the reuse of the same api in a centralized way